prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -99,9 +73,72 @@ class OktaAuthentication(private val minimumLevel: PrincipalLevel = PrincipalLev
+         *
+         * If [requireSenderClaim] is true, then this user must have a claim of the form DHSender_organizationName,
+         * or be an admin.
+         * If [requireSenderClaim] is false,then this user must have a claim of the form DHorganizationName, or
+         * DHSender_organizationName, or be an admin.
+         * [requiredOrganizationName] is the optional organization the caller desires to be associated with.
+         * Note:  underscores and dashes in both claims and [requiredOrganizationName] are treated as identical.
+         * That is, for example, DHa_b-c is the same organization as DHa-b_c.  This is to cover for widespread
+         * inconsistencies between how Okta and Settings handle "-" and "_".
+         */
+        fun authorizeByMembership(
+            claims: AuthenticatedClaims,
+            requiredMinimumLevel: PrincipalLevel,
+            requiredOrganizationName: String?,
+            requireSenderClaim: Boolean = false
+        ): Boolean {
+            @Suppress("UNCHECKED_CAST")
+            val membershipsFromOkta = (claims.jwtClaims[oktaMembershipClaim] as? Collection<String> ?: return false)
+                .filter { !it.isNullOrBlank() }
+                .map { it.replace("-", "_") }
+            // Requirement: User's claims must exactly match one of these strings to be authorized.
+            val requiredMemberships = when (requiredMinimumLevel) {
+                PrincipalLevel.SYSTEM_ADMIN -> listOf(oktaSystemAdminGroup)
+                PrincipalLevel.ORGANIZATION_ADMIN -> {
+                    listOf(
+                        "$oktaGroupPrefix$requiredOrganizationName$oktaAdminGroupSuffix",
+                        oktaSystemAdminGroup
+                    )
+                }
+                PrincipalLevel.USER ->
+                    listOfNotNull(
+                        if (!requireSenderClaim) "$oktaGroupPrefix$requiredOrganizationName" else null,
+                        "$oktaSenderGroupPrefix$requiredOrganizationName",
+                        "$oktaGroupPrefix$requiredOrganizationName$oktaAdminGroupSuffix",
+                        oktaSystemAdminGroup
+                    )
+            }.map { it.replace("-", "_") }
+
+            requiredMemberships.forEach {
+                if (membershipsFromOkta.contains(it)) {
+                    logger.info(
+                        "User ${claims.userName}" +
+                            " memberships $membershipsFromOkta matched requested membership $it"
+                    )
+                    return true
+                }
+            }
+            logger.warn(
+                "User ${claims.userName}" +
+                    " memberships $membershipsFromOkta did NOT match requested $requiredMemberships"
+            )
+            return false
+        }
     }
 
     /**
+     * This is DEPRECATED.  Please stop using this, as it only does Okta auth, and does not allow for
```

Comment:
Were follow-up tickets created for migrating/removing the deprecated logic we should no longer be using?

Reply:
Yes, that's the ticket to move settings, lookuptable and other azure functions over to the new way.